### DISCUSSION TITLE
How to change external useState when mocking component?

### DISCUSSION THREAD

**Discussion by luca-costa-retex** [nuxt/test-utils#721] [Q&A]
I have a component that changes its state by watching a useState that gets changed from outside. How do I unit test it?

**Comment by Rigo-m** [comment]
@danielroe I too am looking into this issue, and I can't quite understand how to test these kind of side effects.
Do you need a quick reproduction?

**Reply by danielroe** [reply]
That would be great! And an issue 🙏

**Reply by Rigo-m** [reply]
Is there a stackblitz ready at hand?

**Reply by danielroe** [reply]
Yes, one of these:

👉 https://stackblitz.com/github/nuxt/test-utils/tree/main/examples/app-vitest
👉 https://stackblitz.com/github/nuxt/test-utils/tree/main/examples/app-jest
👉 https://stackblitz.com/github/nuxt/test-utils/tree/main/examples/module
